djautotask/sync_rest.py

```diff
@@ -781,6 +790,225 @@ def _assign_field_data(self, instance, object_data):
         return instance
 
 
+class ServiceCallSynchronizer(BatchQueryMixin, Synchronizer):
+    client_class = api.ServiceCallsAPIClient
+    model_class = models.ServiceCallTracker
+    condition_field_name = 'companyID'
+    last_updated_field = 'lastModifiedDateTime'
+
+    related_meta = {
+        'companyID': (models.Account, 'account'),
+        'companyLocationID':
+            (models.AccountPhysicalLocation, 'location'),
+        'status': (models.ServiceCallStatus, 'status'),
+        'creatorResourceID': (models.Resource, 'creator_resource'),
+        'canceledByResourceID': (models.Resource, 'canceled_by_resource')
+    }
+
+    @property
+    def active_ids(self):
+        active_ids = models.Account.objects.filter(
+            active=True
+        ).values_list('id', flat=True).order_by(self.lookup_key)
+
+        return active_ids
+
+    def _assign_field_data(self, instance, object_data):
+        instance.id = object_data['id']
+        instance.description = object_data.get('description')
+        instance.duration = Decimal(str(object_data.get('duration')))
+        instance.complete = object_data.get('isComplete')
+        instance.create_date_time = object_data.get('createDateTime')
+        instance.start_date_time = object_data.get('startDateTime')
+        instance.end_date_time = object_data.get('endDateTime')
+        instance.canceled_date_time = object_data.get('canceledDateTime')
+        instance.last_modified_date_time = \
+            object_data.get('lastModifiedDateTime')
+
+        self._set_datetime_attribute(instance, 'create_date_time')
+        self._set_datetime_attribute(instance, 'start_date_time')
+        self._set_datetime_attribute(instance, 'end_date_time')
+        self._set_datetime_attribute(instance, 'canceled_date_time')
+        self._set_datetime_attribute(instance, 'last_modified_date_time')
+
+        self.set_relations(instance, object_data)
+
+        return instance
+
+    def create(self, **kwargs):
+        """
+        Make a request to Autotask to create a ServiceCall.
+        """
+
+        body = {
+            'companyID': kwargs['account'].id,
+            'companyLocationID': kwargs['location'].id,
+            'startDateTime': kwargs['start_date_time'],
+            'endDateTime': kwargs['end_date_time'],
+            'status': kwargs['status'].id,
+            'description': kwargs['description'],
+        }
+        instance = super().create(body)
+        created_item = self.get_single(instance['id'])
```

Comment:
Isn't `instance` already the created record? Do you need to fetch it again with `get_single`? Can't you just pass it into `update_or_create_instance`?

Reply:
Applied. You can now have another look. Thanks~